<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Run scLinaX-multi using the scATAC data and output from scLinaX (Currently, only single sample mode is implemented). — run_scLinaX_multi • scLinaX</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Run scLinaX-multi using the scATAC data and output from scLinaX (Currently, only single sample mode is implemented). — run_scLinaX_multi"><meta name="description" content="Run scLinaX-multi using the scATAC data and output from scLinaX (Currently, only single sample mode is implemented)."><meta property="og:description" content="Run scLinaX-multi using the scATAC data and output from scLinaX (Currently, only single sample mode is implemented)."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">scLinaX</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/scLinaX_preprocessing_example.html">scLinaX_preprocessing_example</a></li>
    <li><a class="dropdown-item" href="../articles/scLinaX_vignette.html">scLinaX_vignette</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Run scLinaX-multi using the scATAC data and output from scLinaX (Currently, only single sample mode is implemented).</h1>

      <div class="d-none name"><code>run_scLinaX_multi.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Run scLinaX-multi using the scATAC data and output from scLinaX (Currently, only single sample mode is implemented).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">run_scLinaX_multi</span><span class="op">(</span></span>
<span>  <span class="va">scLinaX_summary</span>,</span>
<span>  <span class="va">scLinaX_obj</span>,</span>
<span>  <span class="va">ATAC_ASE_data</span>,</span>
<span>  <span class="va">peak_data</span>,</span>
<span>  SNP_DETECTION_DP <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  SNP_DETECTION_MAF <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-sclinax-summary">scLinaX_summary<a class="anchor" aria-label="anchor" href="#arg-sclinax-summary"></a></dt>
<dd><p>Output from the summarize_scLinaX function run with Annotaion=NULL option</p></dd>


<dt id="arg-sclinax-obj">scLinaX_obj<a class="anchor" aria-label="anchor" href="#arg-sclinax-obj"></a></dt>
<dd><p>A list object generated by the <code>run_scLinaX</code> function.</p></dd>


<dt id="arg-atac-ase-data">ATAC_ASE_data<a class="anchor" aria-label="anchor" href="#arg-atac-ase-data"></a></dt>
<dd><p>A dataframe (tibble) containing single-cell chromatin accessibility data. This dataframe should include the following columns:</p><ul><li><p>SNP_ID: SNP identifier</p></li>
<li><p>POS: Genomic position of the SNP (GRCh38)</p></li>
<li><p>REF: Reference allele of the SNP A,T,G,C</p></li>
<li><p>ALT: Alternative allele of the SNP A,T,G,C</p></li>
<li><p>cell_barcode: Cell barcode</p></li>
<li><p>REFcount: Allelic read count of the reference allele</p></li>
<li><p>ALTcount: Allelic read count of the alternative allele</p></li>
<li><p>OTHcount: Allelic read count of the other allele</p></li>
<li><p>Sample_ID: Sample ID</p></li>
</ul></dd>


<dt id="arg-peak-data">peak_data<a class="anchor" aria-label="anchor" href="#arg-peak-data"></a></dt>
<dd><p>A dataframe (tibble) containing peak information called from the single-cell chromatin accessibility data. This dataframe should include the following columns:</p><ul><li><p>start: Genomic position of the start of the peak (1-base)</p></li>
<li><p>end: Genomic position of the end of the peak (1-base)</p></li>
<li><p>Peak_name: Name of the peak</p></li>
<li><p>Gene: Nearest gene to the peak</p></li>
</ul></dd>


<dt id="arg-snp-detection-dp">SNP_DETECTION_DP<a class="anchor" aria-label="anchor" href="#arg-snp-detection-dp"></a></dt>
<dd><p>Threshold for the total allele count (depth) of the SNP in the scATAC data. SNP–Sample pairs with a total allele count of at least "SNP_DETECTION_DP" are used for the analysis. Default: 30.</p></dd>


<dt id="arg-snp-detection-maf">SNP_DETECTION_MAF<a class="anchor" aria-label="anchor" href="#arg-snp-detection-maf"></a></dt>
<dd><p>Threshold for the minor allele count of the scASE data. SNP–Sample pairs with a minor allele ratio between "SNP_DETECTION_MAF" and "1 - SNP_DETECTION_MAF" are used for the analysis. Default: 0.1.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A dataframe (tibble) containing raw per-cell scLinaX-multi results.</p>
<p>This dataframe should include the following columns:</p><ul><li><p>cell_barcode: Cell barcode</p></li>
<li><p>Sample_ID: Sample ID</p></li>
<li><p>SNP_ID: SNP identifier</p></li>
<li><p>CHR: PAR or nonPAR</p></li>
<li><p>POS: Genomic position of the SNP (GRCh38)</p></li>
<li><p>REF: Reference allele of the SNP A,T,G,C</p></li>
<li><p>ALT: Alternative allele of the SNP A,T,G,C</p></li>
<li><p>REFcount: Allelic read count of the reference allele</p></li>
<li><p>ALTcount: Allelic read count of the alternative allele</p></li>
<li><p>OTHcount: Allelic read count of the other allele</p></li>
<li><p>Peak_name: Name of the peak</p></li>
<li><p>Gene: Nearest gene to the peak</p></li>
<li><p>XCI_status: XCI status escape, variable, inactive, unknown</p></li>
<li><p>Gene_class: XCI status combined with CHR information PAR1, nonPAR_escape, nonPAR_variable, nonPAR_inactive, nonPAR_unknown, PAR2</p></li>
<li><p>Xa: Information of the activated X chromosome Allele_A, Allele_B</p></li>
<li><p>A_allele_count, B_allele_count: Total allele count of allele A and B</p></li>
</ul></div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Yoshihiko Tomofuji.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

