<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Run scLinaX-multi using the scATAC data and output from scLinaX (Currently, only single sample mode is implemented)."><title>Run scLinaX-multi using the scATAC data and output from scLinaX (Currently, only single sample mode is implemented). — run_scLinaX_multi • scLinaX</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Run scLinaX-multi using the scATAC data and output from scLinaX (Currently, only single sample mode is implemented). — run_scLinaX_multi"><meta property="og:description" content="Run scLinaX-multi using the scATAC data and output from scLinaX (Currently, only single sample mode is implemented)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scLinaX</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/scLinaX_vignette.html">scLinaX_vignette</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Run scLinaX-multi using the scATAC data and output from scLinaX (Currently, only single sample mode is implemented).</h1>
      
      <div class="d-none name"><code>run_scLinaX_multi.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Run scLinaX-multi using the scATAC data and output from scLinaX (Currently, only single sample mode is implemented).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">run_scLinaX_multi</span><span class="op">(</span></span>
<span>  <span class="va">scLinaX_summary</span>,</span>
<span>  <span class="va">scLinaX_obj</span>,</span>
<span>  <span class="va">ATAC_ASE_data</span>,</span>
<span>  <span class="va">peak_data</span>,</span>
<span>  SNP_DETECTION_DP <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  SNP_DETECTION_MAF <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>scLinaX_summary</dt>
<dd><p>Output from the summarize_scLinaX function run with Annotaion=NULL option</p></dd>


<dt>scLinaX_obj</dt>
<dd><p>A list object generated by the <code>run_scLinaX</code> function.</p></dd>


<dt>ATAC_ASE_data</dt>
<dd><p>A dataframe (tibble) containing single-cell chromatin accessibility data. This dataframe should include the following columns:</p><ul><li><p>SNP_ID: SNP identifier</p></li>
<li><p>POS: Genomic position of the SNP (GRCh38)</p></li>
<li><p>REF: Reference allele of the SNP A,T,G,C</p></li>
<li><p>ALT: Alternative allele of the SNP A,T,G,C</p></li>
<li><p>cell_barcode: Cell barcode</p></li>
<li><p>REFcount: Allelic read count of the reference allele</p></li>
<li><p>ALTcount: Allelic read count of the alternative allele</p></li>
<li><p>OTHcount: Allelic read count of the other allele</p></li>
<li><p>Sample_ID: Sample ID</p></li>
</ul></dd>


<dt>peak_data</dt>
<dd><p>A dataframe (tibble) containing peak information called from the single-cell chromatin accessibility data. This dataframe should include the following columns:</p><ul><li><p>start: Genomic position of the start of the peak (1-base)</p></li>
<li><p>end: Genomic position of the end of the peak (1-base)</p></li>
<li><p>Peak_name: Name of the peak</p></li>
<li><p>Gene: Nearest gene to the peak</p></li>
</ul></dd>


<dt>SNP_DETECTION_DP</dt>
<dd><p>Threshold for the total allele count (depth) of the SNP in the scATAC data. SNP–Sample pairs with a total allele count of at least "SNP_DETECTION_DP" are used for the analysis. Default: 30.</p></dd>


<dt>SNP_DETECTION_MAF</dt>
<dd><p>Threshold for the minor allele count of the scASE data. SNP–Sample pairs with a minor allele ratio between "SNP_DETECTION_MAF" and "1 - SNP_DETECTION_MAF" are used for the analysis. Default: 0.1.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A dataframe (tibble) containing raw per-cell scLinaX-multi results.</p>


<p>This dataframe should include the following columns:</p><ul><li><p>cell_barcode: Cell barcode</p></li>
<li><p>Sample_ID: Sample ID</p></li>
<li><p>SNP_ID: SNP identifier</p></li>
<li><p>CHR: PAR or nonPAR</p></li>
<li><p>POS: Genomic position of the SNP (GRCh38)</p></li>
<li><p>REF: Reference allele of the SNP A,T,G,C</p></li>
<li><p>ALT: Alternative allele of the SNP A,T,G,C</p></li>
<li><p>REFcount: Allelic read count of the reference allele</p></li>
<li><p>ALTcount: Allelic read count of the alternative allele</p></li>
<li><p>OTHcount: Allelic read count of the other allele</p></li>
<li><p>Peak_name: Name of the peak</p></li>
<li><p>Gene: Nearest gene to the peak</p></li>
<li><p>XCI_status: XCI status escape, variable, inactive, unknown</p></li>
<li><p>Gene_class: XCI status combined with CHR information PAR1, nonPAR_escape, nonPAR_variable, nonPAR_inactive, nonPAR_unknown, PAR2</p></li>
<li><p>Xa: Information of the activated X chromosome Allele_A, Allele_B</p></li>
<li><p>A_allele_count, B_allele_count: Total allele count of allele A and B</p></li>
</ul></div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Yoshihiko Tomofuji.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

