<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Run scLinaX analysis for all samples"><title>Run scLinaX analysis for all samples — run_scLinaX • scLinaX</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Run scLinaX analysis for all samples — run_scLinaX"><meta property="og:description" content="Run scLinaX analysis for all samples"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scLinaX</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/scLinaX_vignette.html">scLinaX_vignette</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Run scLinaX analysis for all samples</h1>
      
      <div class="d-none name"><code>run_scLinaX.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Run scLinaX analysis for all samples</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">run_scLinaX</span><span class="op">(</span></span>
<span>  <span class="va">ASE_df</span>,</span>
<span>  <span class="va">XCI_ref</span>,</span>
<span>  <span class="va">QCREF</span>,</span>
<span>  Inactive_Gene_ratio_THR <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  SNP_DETECTION_DP <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  SNP_DETECTION_MAF <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  QC_total_allele_THR <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  HE_allele_cell_number_THR <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  REMOVE_ESCAPE <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  PVAL_THR <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  RHO_THR <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>ASE_df</dt>
<dd><p>A dataframe (tibble) containing single-cell allele-specific expression (scASE) data for all samples. This dataframe should have the following columns:</p><ul><li><p>SNP_ID: SNP identifier</p></li>
<li><p>POS: Genomic position of the SNP (GRCh38)</p></li>
<li><p>REF: Reference allele of the SNP A,T,G,C</p></li>
<li><p>ALT: Alternative allele of the SNP A,T,G,C</p></li>
<li><p>cell_barcode: Cell barcode</p></li>
<li><p>REFcount: Allelic expression of the reference allele</p></li>
<li><p>ALTcount: Allelic expression of the alternative allele</p></li>
<li><p>OTHcount: Allelic expression of the other allele</p></li>
<li><p>Sample_ID: Sample ID</p></li>
<li><p>Gene: Gene annotated to the SNP</p></li>
</ul></dd>


<dt>XCI_ref</dt>
<dd><p>A dataframe (tibble) containing X chromosome inactivation status. This dataframe should have the following two columns:</p><ul><li><p>Gene: Gene name</p></li>
<li><p>XCI_status: XCI status escape, variable, inactive</p></li>
</ul></dd>


<dt>QCREF</dt>
<dd><p>A dataframe generated for reference gene QC (please see the <code>run_RefGeneQC</code> function). QCREF should have the following three columns:</p><ul><li><p>Gene</p></li>
<li><p>Mean_AR_target</p></li>
<li><p>Mean_AR_reference</p></li>
</ul></dd>


<dt>Inactive_Gene_ratio_THR</dt>
<dd><p>Threshold for the ratio of inactive genes</p></dd>


<dt>SNP_DETECTION_DP</dt>
<dd><p>Threshold for the total allele count (depth) of the SNP in the scASE data. SNP–Sample pairs with a total allele count of at least "SNP_DETECTION_DP" are used for the analysis. Default: 30.</p></dd>


<dt>SNP_DETECTION_MAF</dt>
<dd><p>Threshold for the minor allele count of the scASE data. SNP–Sample pairs with a minor allele ratio between "SNP_DETECTION_MAF" and "1 - SNP_DETECTION_MAF" are used for the analysis. Default: 0.1.</p></dd>


<dt>QC_total_allele_THR</dt>
<dd><p>Threshold for the total allele count (depth) of the SNP used for calculating the ratio of expression from Xi.
Note that this count is calculated with cells successfully assigned to the group based on the inactivated X chromosome.
This filter is applied in the final step of scLinax and differs from "SNP_DETECTION_DP". Default: 10.</p></dd>


<dt>HE_allele_cell_number_THR</dt>
<dd><p>Threshold for the number of cells expressing reference SNPs. Candidate reference SNPs expressed in at least "HE_allele_cell_number_THR" cells are used for the analysis. Default: 50.</p></dd>


<dt>REMOVE_ESCAPE</dt>
<dd><p>Boolean:</p><ul><li><p>TRUE (default): Remove ASE profiles of SNPs on escapee genes when calculating correlations between pseudobulk ASE profiles.</p></li>
<li><p>FALSE: Include ASE profiles of SNPs on escapee genes when calculating correlations between pseudobulk ASE profiles.</p></li>
</ul></dd>


<dt>PVAL_THR</dt>
<dd><p>Threshold for P-values and absolute correlation coefficients in the correlation analysis of pseudobulk profiles generated for reference SNPs.</p></dd>


<dt>RHO_THR</dt>
<dd><p>Threshold for absolute correlation coefficients in the correlation analysis of pseudobulk profiles generated for reference SNPs.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A list of objects ($result, $raw_exp_result, $Fail_list) representing the results of scLinaX.
Additional objects representing intermediate results of scLinaX are also included in the list ($clustering_result, $Max_Num_Table_result, $df_snp_summary, $phasing_result).
While users may not typically need to use these intermediate objects directly, they are provided for reference.</p><ul><li><p>$result: A per-sample raw result of scLinaX. This data has redundancy and should be summarized using the <code>summarize_scLinaX</code> function before analysis.</p>
<p>The dataframe ($result) has the following columns:</p><ul><li><p>Sample_ID: Sample ID</p></li>
<li><p>SNP_ID: SNP identifier</p></li>
<li><p>CHR: PAR or nonPAR</p></li>
<li><p>POS: Genomic position of the SNP (GRCh38)</p></li>
<li><p>REF: Reference allele of the SNP A,T,G,C</p></li>
<li><p>ALT: Alternative allele of the SNP A,T,G,C</p></li>
<li><p>Gene: Gene name</p></li>
<li><p>XCI_status: XCI status escape, variable, inactive, unknown</p></li>
<li><p>Gene_class: XCI status combined with CHR information PAR1, nonPAR_escape, nonPAR_variable, nonPAR_inactive, nonPAR_unknown, PAR2</p></li>
<li><p>Used_as_refGene: Whether the Gene was used as a reference gene Yes, No</p></li>
<li><p>Used_as_refSNP: Whether the SNP_ID was used as a reference SNP Yes, No</p></li>
<li><p>Total_A_allele, Total_B_allele: Total allele count of allele A and B</p></li>
<li><p>Total_allele: Total allele count of the SNP</p></li>
<li><p>Expressing_cells: Number of cells expressing the SNP</p></li>
<li><p>minor_allele_ratio: Ratio of the expression from the allele (A, B) with lower expression</p></li>
<li><p>Reference_Gene, Reference_SNP: A list of genes used as reference genes and SNPs</p></li>
<li><p>Num_Reference_Gene, Num_Reference_SNP: Number of reference genes and SNPs</p></li>
<li><p>Reference_Cell_Count: Number of cells showing mono-allelic expression of reference SNPs</p></li>
<li><p>Num_A_cells, Num_B_cells: Number of cells showing mono-allelic expression of A and B alleles of reference SNPs</p></li>
<li><p>Num_Fail_cells: Number of cells showing bi-allelic expression of reference SNPs (should be removed from the analysis)</p></li>
</ul></li>
<li><p>$raw_exp_result: A per-sample raw result of scLinaX.</p>
<p>The dataframe ($raw_exp_result) has the following columns:</p><ul><li><p>cell_barcode: Cell barcode</p></li>
<li><p>Sample_ID: Sample ID</p></li>
<li><p>SNP_ID: SNP identifier</p></li>
<li><p>CHR: PAR or nonPAR</p></li>
<li><p>POS: Genomic position of the SNP (GRCh38)</p></li>
<li><p>REF: Reference allele of the SNP A,T,G,C</p></li>
<li><p>ALT: Alternative allele of the SNP A,T,G,C</p></li>
<li><p>REFcount: Allelic expression of the reference allele</p></li>
<li><p>ALTcount: Allelic expression of the alternative allele</p></li>
<li><p>OTHcount: Allelic expression of the other allele</p></li>
<li><p>Gene: Gene name</p></li>
<li><p>XCI_status: XCI status escape, variable, inactive, unknown</p></li>
<li><p>Gene_class: XCI status combined with CHR information PAR1, nonPAR_escape, nonPAR_variable, nonPAR_inactive, nonPAR_unknown, PAR2</p></li>
<li><p>Used_as_refGene: Whether the Gene was used as a reference gene Yes, No</p></li>
<li><p>Used_as_refSNP: Whether the SNP_ID was used as a reference SNP Yes, No</p></li>
<li><p>Xa: Information of the activated X chromosome Allele_A, Allele_B</p></li>
<li><p>Reference_Gene, Reference_SNP: A list of genes used as reference genes and SNPs</p></li>
</ul></li>
<li><p>$Fail_list: List of samples for which scLinaX analysis failed.</p></li>
<li><p>$clustering_result: Result of the grouping of reference SNPs. Cluster names are in the format Sample_ID<em>cluster</em>cluster_ID.</p></li>
<li><p>$Max_Num_Table_result: A dataframe describing the number of cells.</p></li>
<li><p>$df_snp_summary: An original dataframe from which Spearman correlation between pseudobulk ASE profiles is calculated.</p></li>
<li><p>$phasing_result: Result of the Spearman correlation analysis for pseudobulk ASE profiles.</p></li>
</ul></div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Yoshihiko Tomofuji.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

