<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Run Gene Quality Control (QC) function — run_RefGeneQC • scLinaX</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Run Gene Quality Control (QC) function — run_RefGeneQC"><meta name="description" content="Run Gene Quality Control (QC) function"><meta property="og:description" content="Run Gene Quality Control (QC) function"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">scLinaX</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/scLinaX_preprocessing_example.html">scLinaX_preprocessing_example</a></li>
    <li><a class="dropdown-item" href="../articles/scLinaX_vignette.html">scLinaX_vignette</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Run Gene Quality Control (QC) function</h1>

      <div class="d-none name"><code>run_RefGeneQC.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Run Gene Quality Control (QC) function</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">run_RefGeneQC</span><span class="op">(</span></span>
<span>  <span class="va">ASE_df</span>,</span>
<span>  <span class="va">XCI_ref</span>,</span>
<span>  SNP_DETECTION_DP <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  SNP_DETECTION_MAF <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  SAMPLE_NUM_THR <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  HE_allele_cell_number_THR <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  QC_total_allele_THR <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-ase-df">ASE_df<a class="anchor" aria-label="anchor" href="#arg-ase-df"></a></dt>
<dd><p>A dataframe (tibble) containing single-cell allele-specific expression (scASE) data for all samples. This dataframe should have the following columns:</p><ul><li><p>SNP_ID: SNP identifier</p></li>
<li><p>POS: Genomic position of the SNP (GRCh38)</p></li>
<li><p>REF: Reference allele of the SNP A,T,G,C</p></li>
<li><p>ALT: Alternative allele of the SNP A,T,G,C</p></li>
<li><p>cell_barcode: Cell barcode</p></li>
<li><p>REFcount: Allelic expression of the reference allele</p></li>
<li><p>ALTcount: Allelic expression of the alternative allele</p></li>
<li><p>OTHcount: Allelic expression of the other allele</p></li>
<li><p>Sample_ID: Sample ID</p></li>
<li><p>Gene: Gene annotated to the SNP</p></li>
</ul></dd>


<dt id="arg-xci-ref">XCI_ref<a class="anchor" aria-label="anchor" href="#arg-xci-ref"></a></dt>
<dd><p>A dataframe (tibble) containing X chromosome inactivation status. This dataframe should have the following two columns:</p><ul><li><p>Gene: Gene name</p></li>
<li><p>XCI_status: XCI status escape, variable, inactive</p></li>
</ul></dd>


<dt id="arg-snp-detection-dp">SNP_DETECTION_DP<a class="anchor" aria-label="anchor" href="#arg-snp-detection-dp"></a></dt>
<dd><p>Threshold for the total allele count (depth) of the SNP in the scASE data. SNP–Sample pairs with a total allele count of at least "SNP_DETECTION_DP" are used for the analysis. Default: 30.</p></dd>


<dt id="arg-snp-detection-maf">SNP_DETECTION_MAF<a class="anchor" aria-label="anchor" href="#arg-snp-detection-maf"></a></dt>
<dd><p>Threshold for the minor allele count of the scASE data. SNP–Sample pairs with a minor allele ratio between "SNP_DETECTION_MAF" and "1 - SNP_DETECTION_MAF" are used for the analysis. Default: 0.1.</p></dd>


<dt id="arg-sample-num-thr">SAMPLE_NUM_THR<a class="anchor" aria-label="anchor" href="#arg-sample-num-thr"></a></dt>
<dd><p>Threshold for the sample size used in the calculation of the ratio of expression from Xi.
Genes evaluated in at least <code>SAMPLE_NUM_THR</code> samples are used for the calculation of the ratio of expression from Xi. Default: 3.</p></dd>


<dt id="arg-he-allele-cell-number-thr">HE_allele_cell_number_THR<a class="anchor" aria-label="anchor" href="#arg-he-allele-cell-number-thr"></a></dt>
<dd><p>Threshold for the number of cells expressing reference SNPs. Candidate reference SNPs expressed in at least "HE_allele_cell_number_THR" cells are used for the analysis. Default: 50.</p></dd>


<dt id="arg-qc-total-allele-thr">QC_total_allele_THR<a class="anchor" aria-label="anchor" href="#arg-qc-total-allele-thr"></a></dt>
<dd><p>Threshold for the total allele count (depth) of the SNP used for calculating the ratio of expression from Xi.
Note that this count is calculated with cells successfully assigned to the group based on the inactivated X chromosome.
This filter is applied in the final step of scLinax and differs from "SNP_DETECTION_DP". Default: 10.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A dataframe (tibble) with the following columns:</p><ul><li><p>Gene: Gene name</p></li>
<li><p>Mean_AR_target, SD_AR_target: Mean and standard deviation of the ratio of expression from Xi across other candidate reference genes when the SNPs on the gene were used as references</p></li>
<li><p>Mean_AR_reference, SD_AR_reference: Mean and standard deviation of the ratio of expression from Xi for the gene when SNPs on other candidate reference genes were used as references</p></li>
<li><p>Mean_Total_allele_target, SD_Total_allele_target: Mean and standard deviation of the total allele count across data points when calculating the ARs defined above for <code>target</code>.</p></li>
<li><p>Mean_Total_allele_reference, SD_Total_allele_reference: Mean and standard deviation of the total allele count across data points when calculating the ARs defined above for <code>reference</code>.</p></li>
<li><p>Sample_N_target: Number of samples calculating the ARs defined above for <code>target</code>.</p></li>
<li><p>Sample_N_reference: Number of samples calculating the ARs defined above for <code>reference</code>.</p></li>
<li><p>Count_target: Number of data points calculating the ARs defined above for <code>target</code>.</p></li>
<li><p>Count_reference: Number of data points calculating the ARs defined above for <code>reference</code>.</p></li>
</ul></div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Yoshihiko Tomofuji.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

