<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>scLinaX_preprocessing_example • scLinaX</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="scLinaX_preprocessing_example">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">scLinaX</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/scLinaX_preprocessing_example.html">scLinaX_preprocessing_example</a></li>
    <li><a class="dropdown-item" href="../articles/scLinaX_vignette.html">scLinaX_vignette</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>scLinaX_preprocessing_example</h1>
            
      

      <div class="d-none name"><code>scLinaX_preprocessing_example.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="preprocessing-for-sclinax">Preprocessing for scLinaX<a class="anchor" aria-label="anchor" href="#preprocessing-for-sclinax"></a>
</h2>
<p>scLinaX takes single-cell-level allele-specific expression (or
chromatin accessibility) data with gene annotation as an input. In this
vignette, we will show an example workflow for making an input with
cellsnp-lite and Annovar as described in the scLinaX paper.</p>
</div>
<div class="section level2">
<h2 id="data-preparation">1. Data preparation<a class="anchor" aria-label="anchor" href="#data-preparation"></a>
</h2>
<p>First, prepare the scRNA-seq (or scATAC-seq) dataset for scLinaX
analysis. In this vignette, we use a multiome dataset provided on the
10X website. (<a href="https://www.10xgenomics.com" class="external-link uri">https://www.10xgenomics.com</a>)</p>
<p>Since we only use X choromosome data, we made bam files which include
only reads derived from X chromosomes. Below, we show an example python
script.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Python script</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="co"># Extract the X chromosome</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="co"># Replace the suffix of the barcode (-1) with the data modality (RNA or ATAC)</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co"># Renaming the barcode is actually unnecessary in this case, but it will be necessary if you want to merge multiple runs of the same sample.</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co"># Therefore, we are changing the barcode name here as an example.</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="ex">import</span> pysam</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="ex">import</span> pandas as pd</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="ex">import</span> subprocess</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="va">bam_files</span><span class="op">=</span>[]</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="va">DIR</span><span class="op">=</span><span class="st">'/path/to/DL_data/'</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="va">BAM_LIST</span><span class="op">=</span><span class="pp">[</span><span class="st">'pbmc_granulocyte_sorted_10k_gex_possorted_bam'</span><span class="ss">,</span><span class="st">'pbmc_granulocyte_sorted_10k_atac_possorted_bam'</span><span class="pp">]</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="cf">for</span> BAM <span class="kw">in</span> BAM_LIST:</span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>    <span class="va">input_bam</span><span class="op">=</span>DIR+BAM+<span class="st">'.bam'</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a>    <span class="va">output_bam</span><span class="op">=</span>BAM+<span class="st">'.barcode.renamed.X.bam'</span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a>    <span class="ex">bam_files.append</span><span class="er">(</span><span class="ex">output_bam</span><span class="kw">)</span></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a>    <span class="ex">in_bam</span> = pysam.AlignmentFile<span class="er">(</span><span class="ex">input_bam,</span> <span class="st">'rb'</span><span class="kw">)</span></span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a>    <span class="ex">out_bam</span> = pysam.AlignmentFile<span class="er">(</span><span class="ex">output_bam,</span> <span class="st">'wb'</span>, template=in_bam<span class="kw">)</span></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a>    <span class="cf">if</span><span class="kw">(</span><span class="ex">BAM</span> == <span class="st">'pbmc_granulocyte_sorted_10k_gex_possorted_bam'</span><span class="kw">)</span><span class="bu">:</span></span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a>        <span class="cf">for</span> aln <span class="kw">in</span> in_bam.fetch<span class="er">(</span><span class="st">"chrX"</span><span class="kw">)</span><span class="bu">:</span></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a>            <span class="cf">if</span> <span class="ex">aln.has_tag</span><span class="er">(</span><span class="st">"CB"</span><span class="kw">)</span><span class="bu">:</span></span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a>                <span class="ex">old_bc</span> = aln.get_tag<span class="er">(</span><span class="st">'CB'</span><span class="kw">)</span></span>
<span id="cb1-28"><a href="#cb1-28" tabindex="-1"></a>                <span class="ex">new_bc</span> = old_bc.replace<span class="er">(</span><span class="st">"-1"</span><span class="ex">,</span> <span class="st">"-RNA"</span><span class="kw">)</span></span>
<span id="cb1-29"><a href="#cb1-29" tabindex="-1"></a>                <span class="ex">aln.set_tag</span><span class="er">(</span><span class="st">'CB'</span><span class="ex">,</span> new_bc<span class="kw">)</span></span>
<span id="cb1-30"><a href="#cb1-30" tabindex="-1"></a>                <span class="ex">out_bam.write</span><span class="er">(</span><span class="ex">aln</span><span class="kw">)</span></span>
<span id="cb1-31"><a href="#cb1-31" tabindex="-1"></a></span>
<span id="cb1-32"><a href="#cb1-32" tabindex="-1"></a>    <span class="cf">elif</span><span class="kw">(</span><span class="ex">BAM</span> == <span class="st">'pbmc_granulocyte_sorted_10k_atac_possorted_bam'</span><span class="kw">)</span><span class="bu">:</span></span>
<span id="cb1-33"><a href="#cb1-33" tabindex="-1"></a>        <span class="cf">for</span> aln <span class="kw">in</span> in_bam.fetch<span class="er">(</span><span class="st">"chrX"</span><span class="kw">)</span><span class="bu">:</span></span>
<span id="cb1-34"><a href="#cb1-34" tabindex="-1"></a>            <span class="cf">if</span> <span class="ex">aln.has_tag</span><span class="er">(</span><span class="st">"CB"</span><span class="kw">)</span><span class="bu">:</span></span>
<span id="cb1-35"><a href="#cb1-35" tabindex="-1"></a>                <span class="ex">old_bc</span> = aln.get_tag<span class="er">(</span><span class="st">'CB'</span><span class="kw">)</span></span>
<span id="cb1-36"><a href="#cb1-36" tabindex="-1"></a>                <span class="ex">new_bc</span> = old_bc.replace<span class="er">(</span><span class="st">"-1"</span><span class="ex">,</span> <span class="st">"-ATAC"</span><span class="kw">)</span></span>
<span id="cb1-37"><a href="#cb1-37" tabindex="-1"></a>                <span class="ex">aln.set_tag</span><span class="er">(</span><span class="st">'CB'</span><span class="ex">,</span> new_bc<span class="kw">)</span></span>
<span id="cb1-38"><a href="#cb1-38" tabindex="-1"></a>                <span class="ex">out_bam.write</span><span class="er">(</span><span class="ex">aln</span><span class="kw">)</span></span>
<span id="cb1-39"><a href="#cb1-39" tabindex="-1"></a></span>
<span id="cb1-40"><a href="#cb1-40" tabindex="-1"></a>    <span class="fu">in_bam.close()</span></span>
<span id="cb1-41"><a href="#cb1-41" tabindex="-1"></a>    <span class="fu">out_bam.close()</span></span>
<span id="cb1-42"><a href="#cb1-42" tabindex="-1"></a></span>
<span id="cb1-43"><a href="#cb1-43" tabindex="-1"></a><span class="ex">subprocess.run</span><span class="er">(</span><span class="ex">[</span><span class="st">"samtools"</span><span class="ex">,</span> <span class="st">"merge"</span>, <span class="st">"pbmc_granulocyte_sorted_10k_gex_plus_atac_possorted.barcode.renamed.X.bam"</span>] + bam_files, check=True<span class="kw">)</span></span>
<span id="cb1-44"><a href="#cb1-44" tabindex="-1"></a><span class="ex">subprocess.run</span><span class="er">(</span><span class="ex">[</span><span class="st">"samtools"</span><span class="ex">,</span> <span class="st">"sort"</span>, <span class="st">"-o"</span>, <span class="st">"Multiome_10x.RNA.ATAC.sorted.bam"</span>, <span class="st">"pbmc_granulocyte_sorted_10k_gex_plus_atac_possorted.barcode.renamed.X.bam"</span>], check=True<span class="kw">)</span></span>
<span id="cb1-45"><a href="#cb1-45" tabindex="-1"></a><span class="ex">subprocess.run</span><span class="er">(</span><span class="ex">[</span><span class="st">"samtools"</span><span class="ex">,</span> <span class="st">"index"</span>, <span class="st">"Multiome_10x.RNA.ATAC.sorted.bam"</span>], check=True<span class="kw">)</span></span>
<span id="cb1-46"><a href="#cb1-46" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" tabindex="-1"></a><span class="ex">subprocess.run</span><span class="er">(</span><span class="ex">[</span><span class="st">"samtools"</span><span class="ex">,</span> <span class="st">"sort"</span>, <span class="st">"-o"</span>, <span class="st">"Multiome_10x.RNA.sorted.bam"</span>, <span class="st">"pbmc_granulocyte_sorted_10k_gex_possorted_bam.barcode.renamed.X.bam"</span>], check=True<span class="kw">)</span></span>
<span id="cb1-48"><a href="#cb1-48" tabindex="-1"></a><span class="ex">subprocess.run</span><span class="er">(</span><span class="ex">[</span><span class="st">"samtools"</span><span class="ex">,</span> <span class="st">"index"</span>, <span class="st">"Multiome_10x.RNA.sorted.bam"</span>], check=True<span class="kw">)</span></span>
<span id="cb1-49"><a href="#cb1-49" tabindex="-1"></a></span>
<span id="cb1-50"><a href="#cb1-50" tabindex="-1"></a><span class="ex">subprocess.run</span><span class="er">(</span><span class="ex">[</span><span class="st">"samtools"</span><span class="ex">,</span> <span class="st">"sort"</span>, <span class="st">"-o"</span>, <span class="st">"Multiome_10x.ATAC.sorted.bam"</span>, <span class="st">"pbmc_granulocyte_sorted_10k_atac_possorted_bam.barcode.renamed.X.bam"</span>], check=True<span class="kw">)</span></span>
<span id="cb1-51"><a href="#cb1-51" tabindex="-1"></a><span class="ex">subprocess.run</span><span class="er">(</span><span class="ex">[</span><span class="st">"samtools"</span><span class="ex">,</span> <span class="st">"index"</span>, <span class="st">"Multiome_10x.ATAC.sorted.bam"</span>], check=True<span class="kw">)</span></span></code></pre></div>
<p>We also need lists of cell barcodes to use with cellsnp-lite. In this
example, we used the barcode list generated by the Cell Ranger Arc
software.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">zcat</span> /path/to/DL_data/filtered_feature_bc_matrix/barcodes.tsv.gz <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">sed</span> <span class="st">'s/-1/-RNA/g'</span> <span class="op">&gt;</span> Multiome_10x.RNA.filtered.barcode.list.tsv</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">zcat</span> /path/to/DL_data/filtered_feature_bc_matrix/barcodes.tsv.gz <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="fu">sed</span> <span class="st">'s/-1/-ATAC/g'</span> <span class="op">&gt;</span> Multiome_10x.ATAC.filtered.barcode.list.tsv</span></code></pre></div>
<p>In the case of scATAC-seq analysis, it can be beneficial to remove
duplicate sequences. For example, you can use a flag reported by the
Cell Ranger software.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-F</span> 1024 <span class="at">-b</span> Multiome_10x.ATAC.sorted.bam <span class="op">&gt;</span> Multiome_10x.ATAC.sorted.rmdup.bam</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">samtools</span> index Multiome_10x.ATAC.sorted.rmdup.bam</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-cellsnp-lite">2. Run cellsnp-lite<a class="anchor" aria-label="anchor" href="#run-cellsnp-lite"></a>
</h2>
<p>Now, we are ready to run cellsnp-lite. In this example, we use it in
variant calling mode [2a]. However, if you have genotype data, you can
try the reference-based mode [1a]. You can also specify common variants
defined in other projects (e.g., the 1000 Genome Project) and run it in
reference-based mode, which is faster.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co">#scRNA-seq data info</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="va">barcode_rna</span><span class="op">=</span><span class="st">"Multiome_10x.RNA.filtered.barcode.list.tsv"</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="va">bam_rna</span><span class="op">=</span><span class="st">"Multiome_10x.RNA.sorted.bam"</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="va">UMITAG_rna</span><span class="op">=</span><span class="st">"UB"</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#run cellsnp for RNA</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="ex">cellsnp-lite</span> <span class="at">-s</span> <span class="va">${bam_rna}</span> <span class="dt">\</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>--minCOUNT 20 <span class="at">-b</span> <span class="va">${barcode_rna}</span> <span class="dt">\</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>--UMItag <span class="va">${UMITAG_rna}</span> <span class="at">-p</span> 4 <span class="dt">\</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>--minMAF 0.05 <span class="dt">\</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>--minMAPQ 20 <span class="dt">\</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>--refseq /path/to/refdata-cellranger-GRCh38-3.0.0/fasta/genome.fa <span class="dt">\ </span>#reference used in cell ranger arc</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="ex">--chrom=chrX</span> <span class="at">-O</span> RNA.chrX.snp.call</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="co">#process output for RNA</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="ex">Rscript</span> /path/to/RCODE_process_cellsnp.r <span class="dt">\</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a>RNA.chrX.snp.call/cellSNP.samples.tsv <span class="dt">\</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>RNA.chrX.snp.call/cellSNP.tag.AD.mtx <span class="dt">\</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a>RNA.chrX.snp.call/cellSNP.tag.DP.mtx <span class="dt">\</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a>RNA.chrX.snp.call/cellSNP.tag.OTH.mtx <span class="dt">\</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a>RNA.chrX.snp.call/cellSNP.base.vcf <span class="dt">\</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a>RNA.chrX.snp.call.summary.tsv.gz</span></code></pre></div>
<p>We use custom R script <code>RCODE_process_cellsnp.r</code> to
process the output.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#RCODE_process_cellsnp.r</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org" class="external-link">readr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">argv</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/commandArgs.html" class="external-link">commandArgs</a></span><span class="op">(</span><span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="va">SAMPLE_NAME</span><span class="op">=</span><span class="va">argv</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">ADmtx</span><span class="op">=</span><span class="va">argv</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">DPmtx</span><span class="op">=</span><span class="va">argv</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="va">OTHmtx</span><span class="op">=</span><span class="va">argv</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="va">VCF</span><span class="op">=</span><span class="va">argv</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="va">OUT</span><span class="op">=</span><span class="va">argv</span><span class="op">[</span><span class="fl">6</span><span class="op">]</span></span>
<span></span>
<span><span class="va">sample_name</span><span class="op">&lt;-</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="va">SAMPLE_NAME</span>,col_names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">AD</span><span class="op">&lt;-</span><span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/externalFormats.html" class="external-link">readMM</a></span><span class="op">(</span><span class="va">ADmtx</span><span class="op">)</span></span>
<span><span class="va">DP</span><span class="op">&lt;-</span><span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/externalFormats.html" class="external-link">readMM</a></span><span class="op">(</span><span class="va">DPmtx</span><span class="op">)</span></span>
<span><span class="va">OTH</span><span class="op">&lt;-</span><span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/externalFormats.html" class="external-link">readMM</a></span><span class="op">(</span><span class="va">OTHmtx</span><span class="op">)</span></span>
<span><span class="va">RD</span><span class="op">&lt;-</span><span class="va">DP</span><span class="op">-</span><span class="va">AD</span></span>
<span><span class="va">vcf</span><span class="op">&lt;-</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="va">VCF</span>,comment<span class="op">=</span><span class="st">"#"</span>,col_names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">SNP_INFO</span><span class="op">&lt;-</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="va">vcf</span>,SNP_ID<span class="op">=</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_c.html" class="external-link">str_c</a></span><span class="op">(</span><span class="va">X1</span>,<span class="st">":"</span>,<span class="va">X2</span>,<span class="st">":"</span>,<span class="va">X4</span>,<span class="st">":"</span>,<span class="va">X5</span><span class="op">)</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">SNP_ID</span>,<span class="va">X1</span>,<span class="va">X2</span>,<span class="va">X4</span>,<span class="va">X5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">SNP_INFO</span><span class="op">)</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNP_ID"</span>,<span class="st">"CHR"</span>,<span class="st">"POS"</span>,<span class="st">"REF"</span>,<span class="st">"ALT"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">alt_res</span><span class="op">&lt;-</span><span class="va">AD</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">alt_res</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">sample_name</span><span class="op">$</span><span class="va">X1</span></span>
<span><span class="va">alt_res</span><span class="op">$</span><span class="va">SNP</span><span class="op">&lt;-</span><span class="va">SNP_INFO</span><span class="op">$</span><span class="va">SNP_ID</span></span>
<span></span>
<span><span class="va">ref_res</span><span class="op">&lt;-</span><span class="va">RD</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ref_res</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">sample_name</span><span class="op">$</span><span class="va">X1</span></span>
<span><span class="va">ref_res</span><span class="op">$</span><span class="va">SNP</span><span class="op">&lt;-</span><span class="va">SNP_INFO</span><span class="op">$</span><span class="va">SNP_ID</span></span>
<span></span>
<span><span class="va">oth_res</span><span class="op">&lt;-</span><span class="va">OTH</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">oth_res</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">sample_name</span><span class="op">$</span><span class="va">X1</span></span>
<span><span class="va">oth_res</span><span class="op">$</span><span class="va">SNP</span><span class="op">&lt;-</span><span class="va">SNP_INFO</span><span class="op">$</span><span class="va">SNP_ID</span></span>
<span></span>
<span><span class="va">alt_res</span><span class="op">&lt;-</span><span class="va">alt_res</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">pivot_longer</span><span class="op">(</span>names_to<span class="op">=</span><span class="st">"cell_barcode"</span>,values_to<span class="op">=</span><span class="st">"ALTcount"</span>,<span class="op">-</span><span class="va">SNP</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ALTcount</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ref_res</span><span class="op">&lt;-</span><span class="va">ref_res</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">pivot_longer</span><span class="op">(</span>names_to<span class="op">=</span><span class="st">"cell_barcode"</span>,values_to<span class="op">=</span><span class="st">"REFcount"</span>,<span class="op">-</span><span class="va">SNP</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">REFcount</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="va">oth_res</span><span class="op">&lt;-</span><span class="va">oth_res</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">pivot_longer</span><span class="op">(</span>names_to<span class="op">=</span><span class="st">"cell_barcode"</span>,values_to<span class="op">=</span><span class="st">"OTHcount"</span>,<span class="op">-</span><span class="va">SNP</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">OTHcount</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="va">result</span><span class="op">&lt;-</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">full_join</a></span><span class="op">(</span><span class="va">ref_res</span>,<span class="va">alt_res</span>,by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNP"</span>,<span class="st">"cell_barcode"</span><span class="op">)</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">full_join</a></span><span class="op">(</span><span class="va">oth_res</span>,by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNP"</span>,<span class="st">"cell_barcode"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">result</span><span class="op">&lt;-</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">SNP_INFO</span>,<span class="va">result</span>,<span class="va">alt_res</span>,by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNP_ID"</span><span class="op">=</span><span class="st">"SNP"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">result</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span><span class="op">]</span><span class="op">&lt;-</span><span class="fl">0</span></span>
<span><span class="va">result</span><span class="op">&lt;-</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html" class="external-link">write_tsv</a></span><span class="op">(</span><span class="va">result</span>,<span class="va">OUT</span><span class="op">)</span></span></code></pre></div>
<p>In the case of scATAC-seq data, we generally do not have UMI.
Therefore, we need to make slight adjustments to the configuration.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co">#scATAC-seq data info</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="va">barcode_atac</span><span class="op">=</span><span class="st">"Multiome_10x.ATAC.filtered.barcode.list.tsv"</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="va">bam_atac</span><span class="op">=</span><span class="st">"Multiome_10x.ATAC.sorted.rmdup.bam"</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="va">UMITAG_atac</span><span class="op">=</span><span class="st">"None"</span> <span class="co"># Different from the scRNA-seq</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#run cellsnp for ATAC</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="ex">cellsnp-lite</span> <span class="at">-s</span> <span class="va">${bam_atac}</span> <span class="dt">\</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>--minCOUNT 20 <span class="at">-b</span> <span class="va">${barcode_atac}</span> <span class="dt">\</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>--UMItag <span class="va">${UMITAG_atac}</span> <span class="at">-p</span> 4 <span class="dt">\</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>--minMAF 0.05 <span class="dt">\</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>--minMAPQ 20 <span class="dt">\</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>--refseq /path/to/refdata-cellranger-GRCh38-3.0.0/fasta/genome.fa <span class="dt">\ </span>#reference used in cell ranger arc</span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="ex">--chrom=chrX</span> <span class="at">-O</span> ATAC.chrX.snp.call</span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="co">#process output for ATAC</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="ex">Rscript</span> /path/to/RCODE_process_cellsnp.r <span class="dt">\</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a>ATAC.chrX.snp.call/cellSNP.samples.tsv <span class="dt">\</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a>ATAC.chrX.snp.call/cellSNP.tag.AD.mtx <span class="dt">\</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a>ATAC.chrX.snp.call/cellSNP.tag.DP.mtx <span class="dt">\</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a>ATAC.chrX.snp.call/cellSNP.tag.OTH.mtx <span class="dt">\</span></span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a>ATAC.chrX.snp.call/cellSNP.base.vcf <span class="dt">\</span></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a>ATAC.chrX.snp.call.summary.tsv.gz</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-annovar">3. Run ANNOVAR<a class="anchor" aria-label="anchor" href="#run-annovar"></a>
</h2>
<p>ANNOVAR is a software to annotate various information on genetic
variants. You can download and find the detailed documentation on their
website (<a href="https://annovar.openbioinformatics.org/en/latest/" class="external-link uri">https://annovar.openbioinformatics.org/en/latest/</a>). In
addition to the gene-based annotation, we also added allele frequency
information to the variants for subsequent quality control. Please
download the annotations necessary for your analysis following the
ANNOVAR User Guide (<a href="https://annovar.openbioinformatics.org/en/latest/user-guide/download/" class="external-link uri">https://annovar.openbioinformatics.org/en/latest/user-guide/download/</a>).</p>
<p>Let’s run ANNOVAR for genotype data derived from the scRNA-seq and
scATAC-seq data.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co">#  scRNA-seq</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="fu">zcat</span> RNA.chrX.snp.call.summary.tsv.gz <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="fu">awk</span> <span class="at">-F</span><span class="st">"\t"</span> <span class="st">'NR&gt;1 {print $2"\t"$3"\t"$3"\t"$4"\t"$5"\t"$1}'</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="va">LANG</span><span class="op">=</span>C <span class="fu">sort</span> <span class="kw">|</span> <span class="va">LANG</span><span class="op">=</span>C <span class="fu">uniq</span> <span class="dt">\</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="op">&gt;</span> RNA.chrX.snp.call.summary.annov.input</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="ex">/path/to/annovar/table_annovar.pl</span> <span class="dt">\</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>RNA.chrX.snp.call.summary.annov.input <span class="dt">\</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>/path/to/annovar/humandb/ <span class="at">-buildver</span> hg38 <span class="dt">\</span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>-out RNA.chrX.snp.call <span class="dt">\</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>-remove <span class="at">-protocol</span> refGene,ensGene,avsnp150,ALL.sites.2015_08 <span class="dt">\</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>-operation g,g,f,f <span class="at">-nastring</span> NA <span class="at">-polish</span></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a><span class="co"># scATAC-seq</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a><span class="fu">zcat</span> ATAC.chrX.snp.call.summary.tsv.gz <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a><span class="fu">awk</span> <span class="at">-F</span><span class="st">"\t"</span> <span class="st">'NR&gt;1 {print $2"\t"$3"\t"$3"\t"$4"\t"$5"\t"$1}'</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a><span class="va">LANG</span><span class="op">=</span>C <span class="fu">sort</span> <span class="kw">|</span> <span class="va">LANG</span><span class="op">=</span>C <span class="fu">uniq</span> <span class="dt">\</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a><span class="op">&gt;</span> ATAC.chrX.snp.call.summary.annov.input</span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a><span class="ex">/path/to/annovar/table_annovar.pl</span> <span class="dt">\</span></span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a>ATAC.chrX.snp.call.summary.annov.input <span class="dt">\</span></span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a>/path/to/annovar/humandb/ <span class="at">-buildver</span> hg38 <span class="dt">\</span></span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a>-out ATAC.chrX.snp.call <span class="dt">\</span></span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a>-remove <span class="at">-protocol</span> refGene,ensGene,avsnp150,ALL.sites.2015_08 <span class="dt">\</span></span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a>-operation g,g,f,f <span class="at">-nastring</span> NA <span class="at">-polish</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quality-control">4. Quality control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h2>
<p>As a final step, we apply quality control (QC) to the allele-specific
expression (chromatin accessibility) data based on the Annovar
annotation. Below, we provide an example R script for QC, which filters
based on gene annotation and allele information. We can use
<code>Multiome_RNA_QC_passed_SNP_df.tsv.gz</code> for scLinaX analysis.
Additionally, we can use
<code>Multiome_ATAC_QC_passed_SNP_df.tsv.gz</code> for scLinaX-multi
analysis.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#RCODE_QC.r</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">viridis</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LKremer/ggpointdensity" class="external-link">ggpointdensity</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">argv</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/commandArgs.html" class="external-link">commandArgs</a></span><span class="op">(</span><span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="va">RNA_SNPCELL</span><span class="op">=</span><span class="va">argv</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">RNA_ANNOVAR</span><span class="op">=</span><span class="va">argv</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">ATAC_SNPCELL</span><span class="op">=</span><span class="va">argv</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="va">ATAC_ANNOVAR</span><span class="op">=</span><span class="va">argv</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span></span>
<span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">{</span></span>
<span><span class="co">#Example of input</span></span>
<span><span class="va">RNA_SNPCELL</span><span class="op">=</span><span class="st">"RNA.chrX.snp.call.summary.tsv.gz"</span></span>
<span><span class="va">RNA_ANNOVAR</span><span class="op">=</span><span class="st">"RNA.chrX.snp.call.hg38_multianno.txt"</span> </span>
<span><span class="va">ATAC_SNPCELL</span><span class="op">=</span><span class="st">"ATAC.chrX.snp.call.summary.tsv.gz"</span></span>
<span><span class="va">ATAC_ANNOVAR</span><span class="op">=</span><span class="st">"ATAC.chrX.snp.call.hg38_multianno.txt"</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co">#load raw data</span></span>
<span><span class="va">rna</span><span class="op">&lt;-</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="va">RNA_SNPCELL</span><span class="op">)</span></span>
<span><span class="va">atac</span><span class="op">&lt;-</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="va">ATAC_SNPCELL</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#load annotation data</span></span>
<span><span class="va">load_anno</span><span class="op">&lt;-</span><span class="kw">function</span><span class="op">(</span><span class="va">ANNOVAR</span><span class="op">)</span><span class="op">{</span></span>
<span><span class="va">anno</span><span class="op">&lt;-</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="va">ANNOVAR</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>SNP_ID<span class="op">=</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_c.html" class="external-link">str_c</a></span><span class="op">(</span><span class="va">Chr</span>,<span class="va">Start</span>,<span class="va">Ref</span>,<span class="va">Alt</span>,sep<span class="op">=</span><span class="st">":"</span><span class="op">)</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">SNP_ID</span>,<span class="va">Func.refGene</span>,<span class="va">Gene.refGene</span>,<span class="va">ALL.sites.2015_08</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">anno</span><span class="op">)</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNP_ID"</span>,<span class="st">"Region"</span>,<span class="st">"Gene"</span>,<span class="st">"ALL_Freq"</span><span class="op">)</span></span>
<span><span class="va">anno</span><span class="op">$</span><span class="va">ALL_Freq</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">anno</span><span class="op">$</span><span class="va">ALL_Freq</span><span class="op">)</span><span class="op">]</span><span class="op">&lt;-</span><span class="fl">0</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">anno</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">rna_anno</span><span class="op">&lt;-</span><span class="fu">load_anno</span><span class="op">(</span><span class="va">RNA_ANNOVAR</span><span class="op">)</span></span>
<span><span class="va">atac_anno</span><span class="op">&lt;-</span><span class="fu">load_anno</span><span class="op">(</span><span class="va">ATAC_ANNOVAR</span><span class="op">)</span></span>
<span></span>
<span> </span>
<span><span class="co">#merge raw data + annotation data</span></span>
<span><span class="va">merge_anno</span><span class="op">&lt;-</span><span class="kw">function</span><span class="op">(</span><span class="va">data</span>,<span class="va">anno</span><span class="op">)</span><span class="op">{</span></span>
<span><span class="va">df</span><span class="op">&lt;-</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">data</span>,<span class="va">anno</span>,by<span class="op">=</span><span class="st">"SNP_ID"</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">Gene</span>,pattern<span class="op">=</span><span class="st">";"</span><span class="op">)</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">Gene</span><span class="op">)</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Region</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"intronic"</span>,<span class="st">"UTR5"</span>,<span class="st">"UTR3"</span>,<span class="st">"exonic"</span>,<span class="st">"ncRNA_exonic"</span>,<span class="st">"ncRNA_intronic"</span>,<span class="st">"splicing"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">rna_df</span><span class="op">&lt;-</span><span class="fu">merge_anno</span><span class="op">(</span><span class="va">rna</span>,<span class="va">rna_anno</span><span class="op">)</span></span>
<span><span class="va">atac_df</span><span class="op">&lt;-</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">atac</span>,<span class="va">atac_anno</span>,by<span class="op">=</span><span class="st">"SNP_ID"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">rna_df_QC_pass</span><span class="op">&lt;-</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">rna_df</span>,<span class="va">ALL_Freq</span><span class="op">&gt;</span><span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="va">atac_df_QC_pass</span><span class="op">&lt;-</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">atac_df</span>,<span class="va">ALL_Freq</span><span class="op">&gt;</span><span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html" class="external-link">write_tsv</a></span><span class="op">(</span><span class="va">rna_df_QC_pass</span>,<span class="st">"Multiome_RNA_QC_passed_SNP_df.tsv.gz"</span><span class="op">)</span> <span class="co">#This QC-passed data should be used for the subsequent analyses</span></span>
<span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html" class="external-link">write_tsv</a></span><span class="op">(</span><span class="va">atac_df_QC_pass</span>,<span class="st">"Multiome_ATAC_QC_passed_SNP_df.tsv.gz"</span><span class="op">)</span> <span class="co">#This QC-passed data should be used for the subsequent analyses</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Yoshihiko Tomofuji.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
