<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>scLinaX_vignette • scLinaX</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="scLinaX_vignette">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">scLinaX</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/scLinaX_preprocessing_example.html">scLinaX_preprocessing_example</a></li>
    <li><a class="dropdown-item" href="../articles/scLinaX_vignette.html">scLinaX_vignette</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>scLinaX_vignette</h1>
            
      

      <div class="d-none name"><code>scLinaX_vignette.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="about-sclinax">About scLinaX<a class="anchor" aria-label="anchor" href="#about-sclinax"></a>
</h2>
<p>One of the two X chromosomes of females is silenced through X
chromosome inactivation (XCI) to compensate for the difference in the
dosage between sexes. Among the X-linked genes, several genes escape
from XCI. scLinaX is a computational tool designed for the
quantification of the escape from X chromosome inactivation using
single-cell RNA sequencing data obtained through droplet-based
methodologies.</p>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>First, load the scLinaX package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ytomofuji.github.io/scLinaX/">scLinaX</a></span><span class="op">)</span></span></code></pre></div>
<p>For demo, load the test dataset included within the scLinaX package
as well.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"XCI_ref"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"AIDA_QCREF"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"multiome_ASE_df"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"multiome_Annotation"</span><span class="op">)</span></span></code></pre></div>
<p>If you would like to run scLinaX with your data, please prepare ASE
data and cell annotation data. In the original manuscript, we prepared
ASE data using cellsnp-lite and Annovar. We applied QC, namely variant
filtering based on the allele frequency in the reference panels
(e.g. 1000 genome project dataset). The format should be like below
(Please also see the reference for the dataset).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">multiome_ASE_df</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 10</span></span></span>
<span><span class="co">#&gt;   SNP_ID           POS REF   ALT   cell_barcode REFcount ALTcount OTHcount Gene </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chrX:299800:… <span style="text-decoration: underline;">299</span>800 C     A     AAACCAACAGG…        1        0        0 PLCX…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chrX:299800:… <span style="text-decoration: underline;">299</span>800 C     A     ACAACACTCTA…        1        0        0 PLCX…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chrX:299800:… <span style="text-decoration: underline;">299</span>800 C     A     ACTAACTCATT…        1        0        0 PLCX…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chrX:299800:… <span style="text-decoration: underline;">299</span>800 C     A     ACTGAATGTGA…        1        0        0 PLCX…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> chrX:299800:… <span style="text-decoration: underline;">299</span>800 C     A     CCGCTAAAGCT…        1        0        0 PLCX…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> chrX:299800:… <span style="text-decoration: underline;">299</span>800 C     A     CGTTGCGCACA…        1        0        0 PLCX…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: Sample_ID &lt;chr&gt;</span></span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">multiome_Annotation</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   cell_barcode       Annotation</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> AAACAGCCAATCCCTT-1 CD4 TCM   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> AAACAGCCACACTAAT-1 CD8 Naive </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> AAACAGCCACCAACCG-1 CD8 Naive </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> AAACAGCCAGGATAAC-1 CD4 Naive </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> AAACAGCCAGTTTACG-1 CD4 TCM   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> AAACAGCCATCCAGGT-1 CD16 Mono</span></span></code></pre></div>
<p>Let’s run scLinaX! The result will be a list object including
<code>$reuslt</code> and <code>$raw_exp_result</code> which are used in
the subsequent process.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scLinaX_res</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/run_scLinaX.html">run_scLinaX</a></span><span class="op">(</span>ASE_df<span class="op">=</span><span class="va">multiome_ASE_df</span>,XCI_ref<span class="op">=</span><span class="va">XCI_ref</span>,QCREF<span class="op">=</span><span class="va">AIDA_QCREF</span>,</span>
<span>                             Inactive_Gene_ratio_THR<span class="op">=</span><span class="fl">0.05</span>,SNP_DETECTION_DP<span class="op">=</span><span class="fl">30</span>,SNP_DETECTION_MAF<span class="op">=</span><span class="fl">0.1</span>,QC_total_allele_THR<span class="op">=</span><span class="fl">10</span>,</span>
<span>                             HE_allele_cell_number_THR<span class="op">=</span><span class="fl">50</span>,REMOVE_ESCAPE<span class="op">=</span><span class="cn">TRUE</span>,PVAL_THR<span class="op">=</span><span class="fl">0.01</span>,RHO_THR<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Start processing of sample 1 (total sample number = 1)"</span></span>
<span><span class="co">#&gt; [1] "Finish processing of sample 1 (total sample number = 1)"</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scLinaX_res</span><span class="op">$</span><span class="va">result</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 24</span></span></span>
<span><span class="co">#&gt;   Sample_ID SNP_ID          CHR      POS REF   ALT   Gene  XCI_status Gene_class</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Multiome  chrX:101044782… nonP… 1.01<span style="color: #949494;">e</span>8 G     A     TRMT… inactive   nonPAR_in…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Multiome  chrX:10139201:… nonP… 1.01<span style="color: #949494;">e</span>7 C     G     WWC3  inactive   nonPAR_in…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Multiome  chrX:10144298:… nonP… 1.01<span style="color: #949494;">e</span>7 C     A     WWC3  inactive   nonPAR_in…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Multiome  chrX:101626444… nonP… 1.02<span style="color: #949494;">e</span>8 A     G     ARMC… inactive   nonPAR_in…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Multiome  chrX:101882824… nonP… 1.02<span style="color: #949494;">e</span>8 G     A     ZMAT1 variable   nonPAR_va…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Multiome  chrX:101931416… nonP… 1.02<span style="color: #949494;">e</span>8 G     A     ZMAT1 variable   nonPAR_va…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 15 more variables: Used_as_refGene &lt;chr&gt;, Used_as_refSNP &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Total_A_allele &lt;dbl&gt;, Total_B_allele &lt;dbl&gt;, Total_allele &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Expressing_cells &lt;int&gt;, minor_allele_ratio &lt;dbl&gt;, Reference_Gene &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Num_Reference_Gene &lt;int&gt;, Reference_SNP &lt;chr&gt;, Num_Reference_SNP &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Reference_Cell_Count &lt;int&gt;, Num_A_cells &lt;int&gt;, Num_B_cells &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Num_Fail_cells &lt;int&gt;</span></span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scLinaX_res</span><span class="op">$</span><span class="va">raw_exp_result</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 20</span></span></span>
<span><span class="co">#&gt;   SNP_ID    Sample_ID    POS REF   ALT   cell_barcode REFcount ALTcount OTHcount</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chrX:101… Multiome  1.01<span style="color: #949494;">e</span>8 G     A     ATGGTTATCCT…        1        0        0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chrX:101… Multiome  1.01<span style="color: #949494;">e</span>8 G     A     CAAGTGAAGTT…        1        0        0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chrX:101… Multiome  1.01<span style="color: #949494;">e</span>8 G     A     CACGCTAAGTT…        1        0        0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chrX:101… Multiome  1.01<span style="color: #949494;">e</span>8 G     A     GCTAAGAAGTT…        1        0        0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> chrX:101… Multiome  1.01<span style="color: #949494;">e</span>8 G     A     GCTATTGCAAT…        1        0        0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> chrX:101… Multiome  1.01<span style="color: #949494;">e</span>8 G     A     GTACTTCGTTG…        1        0        0</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 11 more variables: Gene &lt;chr&gt;, XCI_status &lt;chr&gt;, CHR &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Gene_class &lt;fct&gt;, A_allele &lt;dbl&gt;, B_allele &lt;dbl&gt;, Xa &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Used_as_refGene &lt;chr&gt;, Used_as_refSNP &lt;chr&gt;, Reference_SNP &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Reference_Gene &lt;chr&gt;</span></span></span></code></pre></div>
<p>Above results should be summarized using the
<code>summarize_scLinaX</code> function before analysis. If no
Annotation is supplied, this function returns the ratio of expression
from the inactivated X chromosome for all cells. If Annotation is
supplied, this function returns the ratio of expression from the
inactivated X chromosome for each cell annotation.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PBMC_summary</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/summarize_scLinaX.html">summarize_scLinaX</a></span><span class="op">(</span><span class="va">scLinaX_res</span>,QC_total_allele_THR<span class="op">=</span><span class="fl">10</span>,Annotation<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">PBMC_summary</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 24</span></span></span>
<span><span class="co">#&gt;   Reference_SNP   Sample_ID Reference_Cell_Count SNP_ID CHR      POS REF   ALT  </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chrX:47624401:… Multiome                  <span style="text-decoration: underline;">2</span>056 chrX:… nonP… 1.54<span style="color: #949494;">e</span>8 G     A    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chrX:47624401:… Multiome                  <span style="text-decoration: underline;">2</span>056 chrX:… nonP… 2.37<span style="color: #949494;">e</span>7 A     G    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chrX:47624401:… Multiome                  <span style="text-decoration: underline;">2</span>056 chrX:… nonP… 1.10<span style="color: #949494;">e</span>8 T     C    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chrX:47624401:… Multiome                  <span style="text-decoration: underline;">2</span>056 chrX:… PAR1  1.59<span style="color: #949494;">e</span>6 C     T    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> chrX:47624401:… Multiome                  <span style="text-decoration: underline;">2</span>056 chrX:… nonP… 1.58<span style="color: #949494;">e</span>7 C     T    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> chrX:47624401:… Multiome                  <span style="text-decoration: underline;">2</span>056 chrX:… nonP… 1.54<span style="color: #949494;">e</span>8 G     A    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 16 more variables: Gene &lt;chr&gt;, XCI_status &lt;chr&gt;, Gene_class &lt;fct&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Used_as_refGene &lt;chr&gt;, Used_as_refSNP &lt;chr&gt;, Total_A_allele &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Total_B_allele &lt;dbl&gt;, Total_allele &lt;dbl&gt;, Expressing_cells &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   minor_allele_ratio &lt;dbl&gt;, Reference_Gene &lt;chr&gt;, Num_Reference_Gene &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Num_Reference_SNP &lt;int&gt;, Num_A_cells &lt;int&gt;, Num_B_cells &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Num_Fail_cells &lt;int&gt;</span></span></span>
<span></span>
<span></span>
<span><span class="va">cell_type_summary</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/summarize_scLinaX.html">summarize_scLinaX</a></span><span class="op">(</span><span class="va">scLinaX_res</span>,QC_total_allele_THR<span class="op">=</span><span class="fl">10</span>,Annotation<span class="op">=</span><span class="va">multiome_Annotation</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cell_type_summary</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 20</span></span></span>
<span><span class="co">#&gt;   Sample_ID SNP_ID     CHR      POS Major_allele Reference_SNP REF   ALT   Gene </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Multiome  chrX:1014… nonP… 1.01<span style="color: #949494;">e</span>7 B            chrX:4762440… C     A     WWC3 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Multiome  chrX:1019… nonP… 1.02<span style="color: #949494;">e</span>8 A            chrX:4762440… T     G     ZMAT1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Multiome  chrX:1036… nonP… 1.04<span style="color: #949494;">e</span>8 A            chrX:4762440… T     C     MORF…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Multiome  chrX:1036… nonP… 1.04<span style="color: #949494;">e</span>8 A            chrX:4762440… T     C     MORF…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Multiome  chrX:1100… nonP… 1.10<span style="color: #949494;">e</span>8 B            chrX:4762440… G     A     TMEM…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Multiome  chrX:1189… nonP… 1.19<span style="color: #949494;">e</span>8 A            chrX:4762440… A     G     LONR…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 11 more variables: XCI_status &lt;chr&gt;, Gene_class &lt;fct&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Used_as_refGene &lt;chr&gt;, Used_as_refSNP &lt;chr&gt;, Reference_Gene &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Annotation &lt;chr&gt;, Major_allele_count &lt;dbl&gt;, Minor_allele_count &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Total_allele &lt;dbl&gt;, Reference_Cell_Count &lt;int&gt;, minor_allele_ratio &lt;dbl&gt;</span></span></span></code></pre></div>
<p>Let’s check the result of the scLinaX analysis. The
<code>minor_allele_ratio</code> (ratio of the expression from Xi) tend
to high in the escapee genes compared to other nonPAR genes as
expected.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span><span class="va">PBMC_summary</span><span class="op">$</span><span class="va">Gene_class</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">PBMC_summary</span><span class="op">$</span><span class="va">Gene_class</span>,levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PAR1"</span>,<span class="st">"nonPAR_escape"</span>,<span class="st">"nonPAR_variable"</span>,<span class="st">"nonPAR_inactive"</span>,<span class="st">"nonPAR_unknown"</span>,<span class="st">"PAR2"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span><span class="op">&lt;-</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">PBMC_summary</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Gene_class</span>,y<span class="op">=</span><span class="va">minor_allele_ratio</span>,fill<span class="op">=</span><span class="va">Gene_class</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, hjust <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,aspect.ratio<span class="op">=</span><span class="fl">2</span><span class="op">/</span><span class="fl">3</span>,legend.position<span class="op">=</span><span class="st">"none"</span><span class="op">)</span><span class="op">+</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span></span>
<span>    values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PAR1"</span><span class="op">=</span><span class="st">"#ae2d68"</span>,<span class="st">"nonPAR_escape"</span><span class="op">=</span><span class="st">"#ff1700"</span>,<span class="st">"nonPAR_variable"</span><span class="op">=</span><span class="st">"#ffa600"</span>,</span>
<span>    <span class="st">"nonPAR_inactive"</span><span class="op">=</span><span class="st">"#9db300"</span>,<span class="st">"nonPAR_unknown"</span><span class="op">=</span><span class="st">"#666666"</span>,<span class="st">"PAR2"</span><span class="op">=</span><span class="st">"#729efd"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"ratio of the expression from Xi"</span><span class="op">)</span></span>
<span><span class="va">p</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; Loading required package: ggplot2</span></span></code></pre></div>
<p><img src="scLinaX_vignette_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="about-sclinax-multi">About scLinaX-multi<a class="anchor" aria-label="anchor" href="#about-sclinax-multi"></a>
</h2>
<p>XCI induces chromatin-level transcriptional repression on Xi, while
the transcriptionally active chromatin state on Xi can be observed under
the escape from XCI. scLinaX-multi is a computational tool designed for
the quantification of the escape from X chromosome inactivation at the
chromatin accessibility level using single-cell multi-omics (RNA + ATAC)
data such as 10X multiome.</p>
</div>
<div class="section level2">
<h2 id="usage-1">Usage<a class="anchor" aria-label="anchor" href="#usage-1"></a>
</h2>
<p>For demo, load the additional test dataset included within the
scLinaX package.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"multiome_ATAC_ASE_data"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"multiome_peak_data"</span><span class="op">)</span></span></code></pre></div>
<p>Let’s run scLinaX-multi! Note that scLinaX-multi takes the output of
scLinaX as an input. The result will be a dataframe (tibble) containing
raw per-cell scLinaX-multi results. which are used in the subsequent
process.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scLinaX_multi_res</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/run_scLinaX_multi.html">run_scLinaX_multi</a></span><span class="op">(</span><span class="va">PBMC_summary</span>,<span class="va">scLinaX_res</span>,<span class="va">multiome_ATAC_ASE_data</span>,</span>
<span>                              <span class="va">multiome_peak_data</span>,SNP_DETECTION_DP<span class="op">=</span><span class="fl">30</span>,SNP_DETECTION_MAF<span class="op">=</span><span class="fl">0.10</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scLinaX_multi_res</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 17</span></span></span>
<span><span class="co">#&gt;   SNP_ID    Sample_ID    POS REF   ALT   cell_barcode REFcount ALTcount OTHcount</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chrX:101… Multiome  1.01<span style="color: #949494;">e</span>7 G     A     AAACCGCGTCT…        1        0        0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chrX:101… Multiome  1.01<span style="color: #949494;">e</span>7 G     A     AACAGATAGAC…        1        0        0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chrX:101… Multiome  1.01<span style="color: #949494;">e</span>7 G     A     ACTTAGTCAGG…        1        0        0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chrX:101… Multiome  1.01<span style="color: #949494;">e</span>7 G     A     AGCCAGTGTTC…        1        0        0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> chrX:101… Multiome  1.01<span style="color: #949494;">e</span>7 G     A     ATGTGAGAGAC…        1        0        0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> chrX:101… Multiome  1.01<span style="color: #949494;">e</span>7 G     A     ATTGTGATCGC…        1        0        0</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 8 more variables: Xa &lt;chr&gt;, Peak_name &lt;chr&gt;, Gene &lt;chr&gt;, XCI_status &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   CHR &lt;chr&gt;, Gene_class &lt;fct&gt;, A_allele_count &lt;dbl&gt;, B_allele_count &lt;dbl&gt;</span></span></span></code></pre></div>
<p>As in the case of the scLinaX analysis, above results should be
summarized using the <code>summarize_scLinaX_multi</code> function
before analysis. If no Annotation is supplied, this function returns the
ratio of accessible chromatin derived from the inactivated X chromosome
for all cells. If Annotation is supplied, this function returns the
ratio of accessible chromatin derived from the inactivated X chromosome
for each cell annotation.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ATAC_PBMC_summary</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/summarize_scLinaX_multi.html">summarize_scLinaX_multi</a></span><span class="op">(</span><span class="va">scLinaX_multi_res</span>,QC_total_allele_THR<span class="op">=</span><span class="fl">10</span>,Annotation<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ATAC_PBMC_summary</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 13</span></span></span>
<span><span class="co">#&gt;   SNP_ID  Peak_name Sample_ID    POS REF   ALT   Gene  Gene_class A_allele_count</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chrX:1… chrX-101… Multiome  1.01<span style="color: #949494;">e</span>7 G     A     CLCN4 nonPAR_in…              6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chrX:1… chrX-102… Multiome  1.02<span style="color: #949494;">e</span>8 T     C     BEX5  nonPAR_un…             21</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chrX:1… chrX-103… Multiome  1.03<span style="color: #949494;">e</span>8 C     G     BEX4  nonPAR_in…              2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chrX:1… chrX-103… Multiome  1.03<span style="color: #949494;">e</span>8 C     G     BEX3  nonPAR_in…              0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> chrX:1… chrX-103… Multiome  1.04<span style="color: #949494;">e</span>8 G     T     TCEA… nonPAR_in…              4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> chrX:1… chrX-103… Multiome  1.04<span style="color: #949494;">e</span>8 A     G     MORF… nonPAR_in…            174</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4 more variables: B_allele_count &lt;dbl&gt;, Total_allele_count &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   minor_allele_ratio &lt;dbl&gt;, Major_allele &lt;chr&gt;</span></span></span>
<span></span>
<span></span>
<span><span class="va">ATAC_cell_type_summary</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/summarize_scLinaX_multi.html">summarize_scLinaX_multi</a></span><span class="op">(</span><span class="va">scLinaX_multi_res</span>,QC_total_allele_THR<span class="op">=</span><span class="fl">10</span>,Annotation<span class="op">=</span><span class="va">multiome_Annotation</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ATAC_cell_type_summary</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 13</span></span></span>
<span><span class="co">#&gt;   SNP_ID   Sample_ID Annotation    POS REF   ALT   Gene  Gene_class Major_allele</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chrX:10… Multiome  CD14 Mono  1.03<span style="color: #949494;">e</span>8 C     G     BEX3  nonPAR_in… B           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chrX:10… Multiome  CD14 Mono  1.04<span style="color: #949494;">e</span>8 G     T     TCEA… nonPAR_in… B           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chrX:10… Multiome  CD14 Mono  1.04<span style="color: #949494;">e</span>8 A     G     MORF… nonPAR_in… A           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chrX:10… Multiome  CD16 Mono  1.04<span style="color: #949494;">e</span>8 A     G     MORF… nonPAR_in… A           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> chrX:10… Multiome  CD4 Naive  1.04<span style="color: #949494;">e</span>8 A     G     MORF… nonPAR_in… A           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> chrX:10… Multiome  CD4 TCM    1.04<span style="color: #949494;">e</span>8 A     G     MORF… nonPAR_in… A           </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4 more variables: A_allele_count &lt;dbl&gt;, B_allele_count &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Total_allele_count &lt;dbl&gt;, minor_allele_ratio &lt;dbl&gt;</span></span></span></code></pre></div>
<p>Let’s check the result of the scLinaX-multi analysis. The
<code>minor_allele_ratio</code> (ratio of the accessible chromatin from
Xi) tend to high in the escapee genes compared to other nonPAR genes as
expected.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span><span class="va">ATAC_PBMC_summary</span><span class="op">$</span><span class="va">Gene_class</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">ATAC_PBMC_summary</span><span class="op">$</span><span class="va">Gene_class</span>,levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PAR1"</span>,<span class="st">"nonPAR_escape"</span>,<span class="st">"nonPAR_variable"</span>,<span class="st">"nonPAR_inactive"</span>,<span class="st">"nonPAR_unknown"</span>,<span class="st">"PAR2"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span><span class="op">&lt;-</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">ATAC_PBMC_summary</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Gene_class</span>,y<span class="op">=</span><span class="va">minor_allele_ratio</span>,fill<span class="op">=</span><span class="va">Gene_class</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, hjust <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,aspect.ratio<span class="op">=</span><span class="fl">2</span><span class="op">/</span><span class="fl">3</span>,legend.position<span class="op">=</span><span class="st">"none"</span><span class="op">)</span><span class="op">+</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span></span>
<span>    values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PAR1"</span><span class="op">=</span><span class="st">"#ae2d68"</span>,<span class="st">"nonPAR_escape"</span><span class="op">=</span><span class="st">"#ff1700"</span>,<span class="st">"nonPAR_variable"</span><span class="op">=</span><span class="st">"#ffa600"</span>,</span>
<span>    <span class="st">"nonPAR_inactive"</span><span class="op">=</span><span class="st">"#9db300"</span>,<span class="st">"nonPAR_unknown"</span><span class="op">=</span><span class="st">"#666666"</span>,<span class="st">"PAR2"</span><span class="op">=</span><span class="st">"#729efd"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"ratio of the accessible chromatin derived from Xi"</span><span class="op">)</span></span>
<span><span class="va">p</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="scLinaX_vignette_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="qc-of-the-reference-genes">QC of the reference genes<a class="anchor" aria-label="anchor" href="#qc-of-the-reference-genes"></a>
</h2>
<p>In scLinaX, inactivated X chromosome was estimated based on the
allelic expression pattern of the X-linked genes completely subjected to
XCI. We used an annotation in Tukiainen et al., Nature, 2017, with
additional manual curation. To remove the potential escapee genes from
the above list, we performed additional QC based on the scASE data.
Although we supply the <code>AIDA_QCREF</code> object which was
generated from the AIDA dataset, user can also make their own data for
gene QC. Here, we demonstrated the usage of the function for making such
data. Note that we strongly recommend using this function with a large
number of samples. For demonstration, we use this function for a
multiome dataset, a single sample dataset, but this is not recommended
in the actual analysis.</p>
</div>
<div class="section level2">
<h2 id="usage-2">Usage<a class="anchor" aria-label="anchor" href="#usage-2"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Multiome_QCREF</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/run_RefGeneQC.html">run_RefGeneQC</a></span><span class="op">(</span><span class="va">multiome_ASE_df</span>,<span class="va">XCI_ref</span>,SNP_DETECTION_DP<span class="op">=</span><span class="fl">30</span>,SNP_DETECTION_MAF<span class="op">=</span><span class="fl">0.1</span>,SAMPLE_NUM_THR<span class="op">=</span><span class="fl">1</span>,HE_allele_cell_number_THR<span class="op">=</span><span class="fl">50</span>,QC_total_allele_THR<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">Multiome_QCREF</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 13</span></span></span>
<span><span class="co">#&gt;   Gene   Mean_AR_target SD_AR_target Mean_Total_allele_target</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> FTX            0.333            <span style="color: #BB0000;">NA</span>                       15</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> RPGR           0.214            <span style="color: #BB0000;">NA</span>                       14</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> MTMR1          0.167            <span style="color: #BB0000;">NA</span>                       18</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> GAB3           0.105            <span style="color: #BB0000;">NA</span>                       19</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> DIAPH2         0.076<span style="text-decoration: underline;">9</span>           <span style="color: #BB0000;">NA</span>                       13</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> RPL10          0.071<span style="text-decoration: underline;">4</span>           <span style="color: #BB0000;">NA</span>                       14</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 9 more variables: SD_Total_allele_target &lt;dbl&gt;, Sample_N_target &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Count_target &lt;int&gt;, Mean_AR_reference &lt;dbl&gt;, SD_AR_reference &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Mean_Total_allele_reference &lt;dbl&gt;, SD_Total_allele_reference &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Sample_N_reference &lt;int&gt;, Count_reference &lt;int&gt;</span></span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Yoshihiko Tomofuji.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
